### ISSUE TITLE
The best way to define parameters in minisom

### ISSUE THREAD

**Issue by bayan-kabbani** [JustGlowing/minisom#109]
First of all, thank you very much for this really useful and comprehensive MINISOM library for everything we need.
the dataset I'm working on clustering it using MINISOM is:
https://www.kaggle.com/rjmanoj/telecom-customer-churn-prediction
Composed of 7032 samples and 21 features
I found it difficult to choose the right parameters to train the network and get the appropriate quantization error.
In the beginning I chose the dimensions of the map square, starting from x=1 to x=20 depending on the base:
x=int(np.sqrt(5* np.sqrt(data.shape[0]))))
And I tried changing the parameters based on this function:

input_len=data.shape[1]
from hyperopt import Trials, STATUS_OK

space={ 'sig': hp.uniform('sig',0.001, 5), 'learning_rate':hp.uniform('learning_rate',0.001, 5) }
def som_fn(space):
    sig= space['sig']
    learning_rate=space['learning_rate']
    val = MiniSom(x=x,y=y,input_len=input_len,sigma=sig,learning_rate=learning_rate).quantization_error(data)
    print(val)
    return {'loss': val, 'status': STATUS_OK}
trials = Trials()
best = fmin(fn=som_fn,space=space,algo=tpe.suggest,max_evals=100,trials=trials)
print('best: {}'.format(best))
for i  ,trial in enumerate(trials.trials[:2]):
    print(i,trial)

The results were as follows :
20 * 20  gaussian
sigma=1
itration=1000 qe= 3.7  ,itration=20000 qe= 2.5
 20*20 triangle
 sigma: 4.04 learning_rate: 2.06
 itration=40 qe= 3.9

Unfortunately, The quantization error remains large no matter how I change the size of the map. I did not succeed in choosing the appropriate parameters for training the network and getting suitable classes for clients
any suggestion or reference that might help me ...
Regards...

**Comment by JustGlowing** [comment]
hi @bayan-kabbani,

This tool can help making sense of the effect of the parameters: https://share.streamlit.io/justglowing/minisom/dashboard/dashboard.py

I suggest to have a look and replicate that with your data using the examples provided.

Also, optimize the parameters against an appropriate metric. If you have a classification problem, use the F-1 score as the quantization error doesn't reflect classification performances.
